app/src/main/java/com/rkm/tasky/util/security/implementation/DataStoreEncryptor.kt

```diff
@@ -0,0 +1,91 @@
+package com.rkm.tasky.util.security.implementation
+
+import android.security.keystore.KeyGenParameterSpec
+import android.security.keystore.KeyProperties
+import com.google.gson.Gson
+import com.rkm.tasky.util.security.abstraction.Encryptor
+import java.security.KeyStore
+import javax.crypto.Cipher
+import javax.crypto.KeyGenerator
+import javax.crypto.SecretKey
+import javax.crypto.spec.GCMParameterSpec
+import javax.inject.Inject
+
+class DataStoreEncryptor @Inject constructor(): Encryptor{
```

Comment:
Instead of using a dedicated class to manage the encryption of your session data, you can use Android’s `EncryptedSharedPreferences` class.

```
single<SharedPreferences> {
  EncryptedSharedPreferences(
    androidApplication(),
    “auth_pref”,
    MasterKey(androidApplication()),
    EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
    EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
  )
}
```